common/tools/dev-tool/README.md

```diff
@@ -0,0 +1,223 @@
+
+### Command Interface
+
+Every command file's exports must implement the `CommandModule` interface defined in `src/util/commandModule.ts`. The interface requires that every command export a constant `commandInfo` that implements the `CommandInfo` interface defined in the same file. The `CommandInfo` interface specifies the name, description, and options (command-line arguments) of the command. The command module must also export an async handler function as its default export. Two helper functions, `leafCommand` and `subCommand` are provided to assist with development and to provide strong type-checking when
+extending dev-tool.
+
+### Creating a new leaf command
+
+To create a new leaf command in one of the existing sub-command, create a new TypeScript file for that command. Make sure that your module exports the required `commandInfo` and default handler function. When creating a command, use the `leafCommand` helper to get a strongly-typed `options` parameter for your handler.
+
+As an example, we can create a new `hello-world` command under the `dev-tool package` sub-command. The command will print out a string using the many different logging functions. It will accept an argument `--echo <string here>` that specifies the string to be printed.
+
+`src/commands/package/hello-world.ts`
+```typescript
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license
+
+import { createPrinter } from "../../util/printer";
+import { leafCommand } from "../../util/commandBuilder";
+
+const log = createPrinter("hello-world");
+
+export const commandInfo = {
+  name: "hello-world",
+  description:
+    "print a lovely message",
+  options: {
+      echo: {
+          kind: "string",
+          description: "override the message to be printed",
+          default: "Hello world!"
+      }
+  }
+} as const;
+
+export default leafCommand(commandInfo, async (options) => {
+    // Demonstrate the colorized command output.
+    log("Normal:", options.echo);
+    log.success("Success:", options.echo);
+    log.info("Info:", options.echo);
+    log.warn("Warn:", options.echo);
+    log.error("Error:", options.echo);
+    log.debug("Debug:", options.echo);
+
+    return true;
+});
+```
+
+(__Note__: the `as const` after the definition of `commandInfo` is important for the type of `options` in the handler to be inferred as tightly as possible.)
+
+As a last step, add a mapping for the `"hello-world"` command to the sub-command map in `src/commands/package/index.ts`. This will allow the command to resolve:
+
+`src/commands/package/index.ts`
+```typescript
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license
+
+import { subCommand } from "../../util/commandBuilder";
+
+export const commandInfo = {
```

Comment:
This file is just being edited in this context, so these are the actual contents of that file, minus the elided extra keys in the `subCommand`.  I can see your point that the example should maybe be more atomic. I could include just the context of the `subCommand` invocation and show how to edit it only, since it's the only thing that changes.

Reply:
Yeah, I read it and I was like "what's he doing with commandInfo" and realized it's just part of the file, not part of the example.